keep/step/step.py

```diff
@@ -167,28 +172,34 @@ def _run_single(self):
             result = self._run_single_async()
         # else, just run the provider
         else:
-            try:
-                rendered_providers_parameters = {}
-                for parameter in self.provider_parameters:
-                    rendered_providers_parameters[parameter] = self.io_handler.render(
-                        self.provider_parameters[parameter]
+            for curr_retry_count in range(self.__retry_count + 1):
+                try:
+                    rendered_providers_parameters = {}
+                    for parameter in self.provider_parameters:
+                        rendered_providers_parameters[parameter] = self.io_handler.render(
+                            self.provider_parameters[parameter]
+                        )
+
+                    if self.step_type == StepType.STEP:
+                        step_output = self.provider.query(**rendered_value)
+                        self.context_manager.set_step_context(
+                            self.step_id, results=step_output, foreach=self.foreach
+                        )
+                    else:
+                        self.provider.notify(**rendered_value)
+
+                    extra_context = self.provider.expose()
+                    rendered_providers_parameters.update(extra_context)
+                    self.context_manager.set_step_provider_paremeters(
+                        self.step_id, rendered_providers_parameters
                     )
-
-                if self.step_type == StepType.STEP:
-                    step_output = self.provider.query(**rendered_value)
-                    self.context_manager.set_step_context(
-                        self.step_id, results=step_output, foreach=self.foreach
-                    )
-                else:
-                    self.provider.notify(**rendered_value)
-
-                extra_context = self.provider.expose()
-                rendered_providers_parameters.update(extra_context)
-                self.context_manager.set_step_provider_paremeters(
-                    self.step_id, rendered_providers_parameters
-                )
-            except Exception as e:
-                raise StepError(e)
+                except Exception as e:
+                    if curr_retry_count == self.__retry_count:
+                        raise StepError(e)
+                    else:
+                        self.logger.info(
```

Comment:
anyway it worth adding the original exception so we will have some trace (str(e))